decidim-core/spec/factories.rb

```diff
@@ -108,8 +108,27 @@
       file { test_file("Exampledocument.odt", "application/vnd.oasis.opendocument") }
     end
   end
+
+  factory :feature, class: Decidim::Feature do
+    name { Decidim::Faker::Localized.sentence(3) }
+    participatory_process
+    feature_type "dummy"
+  end
+
+  factory :component, class: Decidim::Component do
+    name { Decidim::Faker::Localized.sentence(3) }
+    component_type { "dummy" }
+
+    step do
+      create(:participatory_process_step,
```

Comment:
This should be `build` instead of `create`. If you need the step to be persisted before creating the component, you can do somthing like:
``` ruby
before(:create) do |component|
  component.step.save! unless component.step.persisted?
end
```